clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc

```diff
@@ -2053,26 +2053,26 @@ inline ::std::pair<Iter1, Iter1> equal_range(_ExecutionPolicy &&policy,
 // DplExtrasVector|is_iterator
 // DPCT_DEPENDENCY_END
 // DPCT_CODE
-template <typename _ExecutionPolicy, typename key_t, typename key_out_t, 
+template <typename _ExecutionPolicy, typename key_t, typename key_out_t,
           typename OffsetIteratorT>
 inline ::std::enable_if_t<dpct::internal::is_iterator<key_t>::value &&
                           dpct::internal::is_iterator<key_out_t>::value>
 segmented_reduce_argmin(_ExecutionPolicy &&policy, key_t keys_in,
-                        key_out_t keys_out, ::std::int64_t n,
-                        ::std::int64_t nsegments, OffsetIteratorT begin_offsets,
+                        key_out_t keys_out, ::std::int64_t nsegments,
+                        OffsetIteratorT begin_offsets,
                         OffsetIteratorT end_offsets) {
   policy.queue().submit([&](sycl::handler &cgh) {
     cgh.parallel_for(nsegments, [=](sycl::id<1> i) {
       ::std::int64_t segment_begin = begin_offsets[i];
       ::std::int64_t segment_length =
-          ::std::min(n, (::std::int64_t)end_offsets[i]) - segment_begin;
+          ((::std::int64_t)end_offsets[i]) - segment_begin;
       if (segment_length <= 0) {
         *(keys_out + i) = dpct::key_value_pair(
-            ptrdiff_t(1),
-            ::std::numeric_limits<
-                typename ::std::iterator_traits<key_t>::value_type>::max());
+            1, ::std::numeric_limits<
+                   typename ::std::iterator_traits<key_t>::value_type>::max());
       } else {
-        dpct::arg_index_input_iterator arg_index(keys_in + segment_begin);
+        dpct::arg_index_input_iterator<decltype(keys_in), int> arg_index(
+            keys_in + segment_begin);
         *(keys_out + i) = *::std::min_element(
```

Comment:
Seems we are missing an array subscript operator in `dpct::arg_index_input_iterator`, can we add `operator[]`? And then here may become more simply. WDYT?